haystack_experimental/components/routers/document_length_router.py

```diff
@@ -0,0 +1,82 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, List
+
+from haystack import component
+from haystack.dataclasses import Document
+
+
+@component
+class DocumentLengthRouter:
+    """
+    Categorizes documents based on the length of the `content` field and routes them to the appropriate output.
+
+    A common use case for DocumentLengthRouter is handling documents obtained from PDFs that contain non-text
+    content, such as scanned pages or images. This component can detect empty or low-content documents and route them to
+    components that perform OCR, generate captions, or compute image embeddings.
+
+    ### Usage example
+
+    ```python
+    from haystack_experimental.components.routers import DocumentLengthRouter
+    from haystack.dataclasses import Document
+
+    docs = [
+        Document(content="Short"),
+        Document(content="Long document "*20),
+    ]
+
+    router = DocumentLengthRouter(threshold=10)
+
+    result = router.run(documents=docs)
+    print(result)
+
+    # {
+    #     "short_documents": [Document(content="Short", ...)],
+    #     "long_documents": [Document(content="Long document ...", ...)],
+    # }
+    """
+
+    def __init__(
+        self,
+        *,
+        threshold: int = 10,
```

Comment:
I was going to say maybe we could make this `Optional[int]` so setting this to None means we only check the None-ness of a Document's content, but this can basically be achieved with setting threshold to 0 as well right?

Reply:
I guess we could also leave as is and explain to users you need to set threshold to -1 to only get None docs. I think I'd be fine with either approach.